### PR TITLE
Add EncodeConcatenatedHashes with test.

### PR DESCRIPTION
**Pull request by chappjc**
For https://github.com/decred/dcrrpcclient/pull/39

### CODE REVIEW

**Comment by chappjc** [comment]
@jrick I know what you mean about the test and I went through several options on this.  Because the concatenated hash string must be non-byte reversed, this is the simplest way to make the input.  The expected output could be written like this (as in `TestDecodeConcatenatedHashes`):  https://github.com/chappjc/dcrd/blob/6981068a68766b4d5b052b49525482dfcacdb02f/dcrjson/parse_test.go#L75
which also is generating the result in a way.
Or would `[]byte{0x... ...}` be preferable like the votebits tests?  Seems like `TestDecodeConcatenatedHashes` would need to be rewritten too.

**Comment by chappjc** [comment]
Or rather `[]chainhash.Hash{{...}, {...}, {...}, ...}`
Is that preferable to make the input?

**Comment by jrick** [comment]
Yeah I would use a slice of chainhash.Hash objects.  Those can be created by calling decodeHash just like the other test does (panics aren't an issue here, the data is hardcoded).  Then just pass that to the function being tested and compare against hard coded expected results.

**Comment by chappjc** [comment]
Without even `decodeHash`.  :grin:   Hardcoded both in and out.

**Comment by jrick** [comment]
decodeHash exists for a reason (readability). please use that.

**Comment by chappjc** [comment]
Done, but marginally less circular than the original test IMO.  Again, seems like `TestDecodeConcatenatedHashes` needs work by the same token.

**Comment by chappjc** [comment]
Comments were slightly off after input/output update.

File: dcrjson/parse_test.go [hunk 1]
```diff
@@ -23,6 +24,39 @@ func decodeHash(reversedHash string) chainhash.Hash {
...
+	hashLen := hex.EncodedLen(len(hashSlice[0]))
+
+	// Expected output. The string representations of the underlying byte arrays
+	// in the input []chainhash.Hash
+	blockHashes := []string{
+		"80d9212bf4ceb066ded2866b39d4ed89e0ab60f335c11df8e7bf85d9c35c8e29",
+		"b926d1870d6f88760a8b10db0d4439e5cd74f3827fd4b6827443000000000000",
+		"badcb8e5c1e895e8e8fef8d3425fa0bfe9d28fdbf72f871910c4000000000000",
+	}
+	concatenatedHashes := strings.Join(blockHashes, "")
```
Comment (chappjc) [inline dcrjson/parse_test.go]: Really don't need a []string, but it seems to parallel input data more clearly this way.

**Pull request merged**